runtime/vm/jvminit.c

```diff
@@ -8205,6 +8207,22 @@ vmHookAnonClassesUnload(J9HookInterface** hook, UDATA eventNum, void* eventData,
 }
 #endif /* JAVA_SPEC_VERSION >= 22 */
 
+static void
+vmHookAnonClassesUnloadCleanupJNIIds(J9HookInterface** hook, UDATA eventNum, void* eventData, void* userData)
+{
+	J9VMAnonymousClassesUnloadEvent * unloadedEvent = (J9VMAnonymousClassesUnloadEvent *)eventData;
+	for (J9Class* j9clazz = unloadedEvent->anonymousClassesToUnload; j9clazz; j9clazz = j9clazz->gcLink) {
+		void **jniIDs = j9clazz->jniIDs;
+		if (NULL != jniIDs) {
+			UDATA size = J9VM_SIZEOF_CLASS_JNIID_TABLE(j9clazz->romClass);
```

Comment:
The macro appears to be size in bytes, not entries. Also, this code should look at the flags to see what behaviour to follow.